### PR TITLE
[cmd/opampsupervisor] Support `AcceptsRestartCommand` Capability

### LINKED ISSUE #21077: [cmd/opampsupervisor] Handle restart command
### Component(s)

_No response_

### Is your feature request related to a problem? Please describe.

The OpAMP specification defines a restart command that can be issued by servers to agents. The Supervisor can be handled either by restarting itself upon receiving these requests, or by restarting the Collector.

### Describe the solution you'd like

Document the intended approach in the Supervisor design document, and implement that approach in the Supervisor. We will likely just want to have the Supervisor restart the Collector without processing config updates upon receiving this message.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by srikanthccv**
**Description:** <Describe what has changed.>

Add restart command support to opamp supervisor capabilities.

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21077

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21077

**Testing:** Added integration test

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by tigrannajaryan — COMMENTED** [review]
Looks good to me. Should this be marked ready for review?

**Comment by srikanthccv** [comment]
I will add the CHANGELOG entry and mark it ready for review.

File: cmd/opampsupervisor/supervisor/supervisor.go [hunk 1]
```diff
@@ -604,6 +608,20 @@ func (s *Supervisor) recalcEffectiveConfig() (configChanged bool, err error) {
 	return configChanged, nil
 }
 
+func (s *Supervisor) handleRestartCommand() error {
+	err := s.commander.Restart(context.Background())
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply cmd/opampsupervisor/supervisor/supervisor.go:612]: Ideally, this should be reported to Server here https://github.com/open-telemetry/opamp-go/blob/main/client/internal/receivedprocessor.go#L59. But there is no clear way to send the status for command back to Server https://github.com/open-telemetry/opamp-spec/issues/164.

**Comment by srikanthccv** [comment]
@evan-bradley / @tigrannajaryan  can you reopen this?

File: cmd/opampsupervisor/supervisor/supervisor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (evan-bradley) [reply cmd/opampsupervisor/supervisor/supervisor.go:713]: Makes sense, thanks. Let's address that after the spec issue is resolved.

**Review by evan-bradley — APPROVED** [review]

**Comment by srikanthccv** [comment]
I see that the test is failing. I will fix that. I suspect the fact that the restart command is not distinguished from the unexpected restart might be the reason.

**Pull request merged**